hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/AccountReadableKVState.java

```diff
@@ -0,0 +1,243 @@
+            TokenAccountRepository tokenAccountRepository,
+            AccountBalanceRepository accountBalanceRepository) {
+        super(KEY);
+        this.commonEntityAccessor = commonEntityAccessor;
+        this.nftAllowanceRepository = nftAllowanceRepository;
+        this.nftRepository = nftRepository;
+        this.tokenAllowanceRepository = tokenAllowanceRepository;
+        this.cryptoAllowanceRepository = cryptoAllowanceRepository;
+        this.tokenAccountRepository = tokenAccountRepository;
+        this.accountBalanceRepository = accountBalanceRepository;
+    }
+
+    @Override
+    protected Account readFromDataSource(@Nonnull AccountID key) {
+        final var timestamp = ContractCallContext.get().getTimestamp();
+        return commonEntityAccessor
+                .get(key, timestamp)
+                .map(entity -> accountFromEntity(entity, timestamp))
+                .orElse(null);
+    }
+
+    @Override
+    protected @Nonnull Iterator<AccountID> iterateFromDataSource() {
+        return Collections.emptyIterator();
+    }
+
+    @Override
+    public long size() {
+        return 0;
+    }
+
+    private Account accountFromEntity(Entity entity, final Optional<Long> timestamp) {
+        var tokenAccountBalances = getNumberOfAllAndPositiveBalanceTokenAssociations(entity.getId(), timestamp);
+
+        return Account.newBuilder()
+                .accountId(convertCanonicalAccountIdFromEntity(entity))
+                .alias(
+                        entity.getEvmAddress() != null && entity.getEvmAddress().length > 0
+                                ? Bytes.wrap(entity.getEvmAddress())
+                                : Bytes.EMPTY)
+                .key(parseKey(entity.getKey()))
+                .expirationSecond(TimeUnit.SECONDS.convert(entity.getEffectiveExpiration(), TimeUnit.NANOSECONDS))
+                .tinybarBalance(getAccountBalance(entity, timestamp))
+                .memo(entity.getMemo())
+                .deleted(Optional.ofNullable(entity.getDeleted()).orElse(false))
+                .receiverSigRequired(entity.getReceiverSigRequired() != null && entity.getReceiverSigRequired())
+                .numberOwnedNfts(getOwnedNfts(entity.getId(), timestamp))
+                .maxAutoAssociations(Optional.ofNullable(entity.getMaxAutomaticTokenAssociations())
+                        .orElse(0))
+                .numberAssociations(() -> tokenAccountBalances.get().all())
+                .smartContract(CONTRACT.equals(entity.getType()))
+                .numberPositiveBalances(() -> tokenAccountBalances.get().positive())
+                .ethereumNonce(entity.getEthereumNonce() != null ? entity.getEthereumNonce() : 0L)
+                .autoRenewAccountId(convertEncodedEntityIdToAccountID(entity.getAutoRenewAccountId()))
+                .autoRenewSeconds(
+                        entity.getAutoRenewPeriod() != null ? entity.getAutoRenewPeriod() : DEFAULT_AUTO_RENEW_PERIOD)
```

Comment:
Avoid repeating the variable without having to allocate a temporary Optional.

```suggestion
                .autoRenewSeconds(Objects.requireNonNullElse(entity.getAutoRenewPeriod(), DEFAULT_AUTO_RENEW_PERIOD))
```